### PR TITLE
move demos out of / and make them compile

### PR DESCRIPTION
**Pull request by matklad**
Will add this to CI in a separate PR

### CODE REVIEW

File: src/config.zig [hunk 1]
```diff
@@ -1,8 +1,10 @@
 const builtin = @import("builtin");
 const std = @import("std");
 
-const build_options = @import("tigerbeetle_build_options");
 const root = @import("root");
+// Allow setting build-time config either via `build.zig` `Options`, or via a stuct in the root file.
+const build_options =
+    if (@hasDecl(root, "vsr_options")) root.vsr_options else @import("vsr_options");
```
Comment (matklad) [inline src/config.zig:7]: Rationale for rename: it's `std_options`, feels better to follow the suite .

File: src/tigerbeetle/main.zig [hunk 2]
```diff
@@ -6,7 +6,7 @@ const mem = std.mem;
 const os = std.os;
 const log_main = std.log.scoped(.main);
 
-const build_options = @import("tigerbeetle_build_options");
+const build_options = @import("vsr_options");
```
Comment (matklad) [inline src/tigerbeetle/main.zig:9]: Here, I think ideally we'd use a different options struct -- we only need this for commit hash, and that feels like it doesn't belong to vsr_options. Splitting the options into two feels a bit out of scope.

**Comment by matklad** [comment]
Added stuff to CI in this PR cc @eatonphil

File: src/config.zig [hunk 3]
```diff
@@ -1,8 +1,10 @@
 const builtin = @import("builtin");
 const std = @import("std");
 
-const build_options = @import("tigerbeetle_build_options");
 const root = @import("root");
+// Allow setting build-time config either via `build.zig` `Options`, or via a stuct in the root file.
```
Comment (sentientwaffle) [inline src/config.zig]:
```suggestion
// Allow setting build-time config either via `build.zig` `Options`, or via a struct in the root file.
```

**Review by sentientwaffle — DISMISSED** [review]
Other than the typo it LGTM (feel free to merge as-is).

**Comment by matklad** [comment]
Should’ve merged as is, committing suggestion dismissed review :(

**Comment by matklad** [comment]
Ah, no, there weren’t an approving review, so I am not in a position to merge as is, as our benevolent automation requires a green checkmark

**Review by sentientwaffle — DISMISSED** [review]

**Comment by matklad** [comment]
@sentientwaffle could you r+?

**Comment by sentientwaffle** [comment]
bors r+

I'm glad you reminded me, I was about to Approve it from muscle memory :stuck_out_tongue:

**Review by sentientwaffle — DISMISSED** [review]

**Comment by sentientwaffle** [comment]
:facepalm: 

bors r+

**Comment by sentientwaffle** [comment]
Probably need to bump the `version` in the node client's `package.json`.

**Comment by matklad** [comment]
Bumped! Now that we have bors, we'd have to fix that though, as it'll become annoying in no time

**Review by sentientwaffle — APPROVED** [review]
bors r+

**Comment by matklad** [comment]
Yeah, one of the limitations of bors is that you can't have flaky tests :D

**Comment by matklad** [comment]
let's re-try this while we are sorting .net issues

bors r=sentientwaffle

(bors has build-in support for impersonating someone)

**Comment by sentientwaffle** [comment]
bors r+

Retrying again, atop https://github.com/tigerbeetledb/tigerbeetle/pull/392
